File: backoffice/src/shared/infrastructure/events/DomainEventsInformation.py
"""
 *
 * Libraries 
 *
"""

from .DomainEventInformation import DomainEventInformation

"""
 *
 * Class
 *
"""

class DomainEventsInformation:

    """
     *
     * Attributes 
     *
    """

    __eventsInformation : list[DomainEventInformation]

    """
     *
     * Methods 
     *
    """

    def __init__( self, eventsInformation : list[DomainEventInformation] ) -> object:
        self.__eventsInformation = eventsInformation

    def getAll( self ) -> list[DomainEventInformation]:
        return self.__eventsInformation

    def findByEventName( self, name : str ) -> DomainEventInformation:
        # Variables
        result : DomainEventInformation
        # Code
        result = None
        for eventInformation in self.__eventsInformation:
            if eventInformation.getEventName() == name:
                result = eventInformation
                break
        return result

File: preparation/src/shared/infrastructure/events/DomainEventsInformation.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.infrastructure {

    public class DomainEventsInformation {

        /* 
         * 
         * Attributes
         *
        */

        private List<DomainEventInformation> eventsInformation;

        /* 
         * 
         * Methods
         *
        */

        public DomainEventsInformation( 
            List<DomainEventInformation> eventsInformation 
        ) {
            this.eventsInformation = eventsInformation;
        }

        public List<DomainEventInformation> getAll() {
            return eventsInformation;
        }

        public DomainEventInformation findByEventName( string name ) {
            // Variables
            DomainEventInformation result;
            // Code
            result = null;
            foreach( DomainEventInformation eventInformation in eventsInformation ) {
                if( eventInformation.getEventName() == name ) {
                    result = eventInformation;
                }
            }
            return result;
        }

    }
}
